examples/pod_portforward_hyper_http.rs

```diff
@@ -0,0 +1,91 @@
+#[macro_use] extern crate log;
+
+use futures::{StreamExt, TryStreamExt};
+use k8s_openapi::api::core::v1::Pod;
+
+use kube::{
+    api::{Api, DeleteParams, ListParams, PostParams, WatchEvent},
+    Client, ResourceExt,
+};
+
+use hyper::{body, client, Body, Request};
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    std::env::set_var("RUST_LOG", "info,kube=debug");
+    env_logger::init();
+    let client = Client::try_default().await?;
+    let namespace = std::env::var("NAMESPACE").unwrap_or_else(|_| "default".into());
+
+    let p: Pod = serde_json::from_value(serde_json::json!({
+        "apiVersion": "v1",
+        "kind": "Pod",
+        "metadata": { "name": "example" },
+        "spec": {
+            "containers": [{
+                "name": "nginx",
+                "image": "nginx",
+            }],
+        }
+    }))?;
+
+    let pods: Api<Pod> = Api::namespaced(client, &namespace);
+    // Stop on error including a pod already exists or is still being deleted.
+    pods.create(&PostParams::default(), &p).await?;
+
+    // Wait until the pod is running, otherwise we get 500 error.
+    let lp = ListParams::default().fields("metadata.name=example").timeout(10);
+    let mut stream = pods.watch(&lp, "0").await?.boxed();
+    while let Some(status) = stream.try_next().await? {
+        match status {
+            WatchEvent::Added(o) => {
+                info!("Added {}", o.name());
+            }
+            WatchEvent::Modified(o) => {
+                let s = o.status.as_ref().expect("status exists on pod");
+                if s.phase.clone().unwrap_or_default() == "Running" {
+                    break;
+                }
+            }
+            _ => {}
+        }
+    }
+
+    let mut pf = pods.portforward("example", &[80]).await?;
+    let mut ports = pf.ports().unwrap();
+    let port = ports[0].stream().unwrap();
+
+    let (mut sender, connection) = client::conn::handshake(port).await?;
+
+    // spawn a task to poll the connection and drive the HTTP state
+    tokio::spawn(async move {
+        if let Err(e) = connection.await {
+            eprintln!("Error in connection: {}", e);
+        }
+    });
```

Comment:
i feel this needs a comment: we already have one task to drive the message machinery in kube-client now, and it might not be clear why we need another task to drive the connection

Reply:
I'm not sure what to write. The comment is from [`hyper::client::conn` example](https://docs.rs/hyper/latest/hyper/client/conn/index.html).

This task drives the HTTP state for IO (`impl AsyncRead + AsyncWrite` from `Port`'s `stream()`), so we can send an HTTP request and receive a response instead of writing and reading bytes like the original example. The message loop in `potforward` simply forwards the bytes to the server by prefixing with the associated port, and doesn't know how the bytes were written.

Reply:
Yeah, it's not a lot of elaborating documentation. Have suggested a minor edit to group the lines that are related to hyper. Not a big deal this though.